tests/phpunit/integration/Core/Permissions/PermissionsTest.php

```diff
@@ -192,8 +194,108 @@ public function test_get_capabilities() {
 			Permissions::VIEW_DASHBOARD,
 			Permissions::VIEW_MODULE_DETAILS,
 			Permissions::MANAGE_OPTIONS,
+			Permissions::VIEW_SHARED_DASHBOARD,
 		);
 
 		$this->assertEqualSets( $capabilities, Permissions::get_capabilities() );
 	}
+
+	public function test_dashboard_sharing_capabilities() {
+		$disable_feature = $this->enable_feature( 'dashboardSharing' );
+
+		$context                  = new Context( GOOGLESITEKIT_PLUGIN_MAIN_FILE );
+		$auth                     = new Authentication( $context );
+		$contributor              = self::factory()->user->create_and_get( array( 'role' => 'contributor' ) );
+		$contributor_user_options = new User_Options( $context, $contributor->ID );
+		$author                   = self::factory()->user->create_and_get( array( 'role' => 'author' ) );
+		$author_user_options      = new User_Options( $context, $author->ID );
+
+		$settings              = new Module_Sharing_Settings( new Options( new Context( GOOGLESITEKIT_PLUGIN_MAIN_FILE ) ) );
+		$test_sharing_settings = array(
+			'analytics'      => array(
+				'sharedRoles' => array( 'contributor' ),
+				'management'  => 'all_admins',
+			),
+			'search-console' => array(
+				'management' => 'owner',
+			),
+		);
+		$settings->set( $test_sharing_settings );
+
+		$permissions = new Permissions( $context, $auth );
+		$permissions->register();
+
+		// Make sure SiteKit is setup.
+		$this->fake_proxy_site_connection();
+		add_filter( 'googlesitekit_setup_complete', '__return_true', 100 );
+		$this->assertTrue( $auth->is_setup_completed() );
+
+		// Test user should have at least one sharedRole and the shared_dashboard_splash
+		// item dismissed to VIEW_SHARED_DASHBOARD.
+		$this->assertFalse( user_can( $author, Permissions::VIEW_SHARED_DASHBOARD ) );
+		$this->assertFalse( user_can( $contributor, Permissions::VIEW_SHARED_DASHBOARD ) );
+		$dismissed_items = new Dismissed_Items( $contributor_user_options );
+		$dismissed_items->add( 'shared_dashboard_splash', 0 );
+		$this->assertTrue( $dismissed_items->is_dismissed( 'shared_dashboard_splash' ) );
```

Comment:
I think this line can be removed, we don't need to test against the behaviour of another class here.
```suggestion
```

Reply:
True - was just being too cautious and over-checking pre-conditions 😄